### ISSUE TITLE
avm1: In-built methods can be overriden

### ISSUE THREAD

**Issue by Toad06** [ruffle-rs/ruffle#5827]
[Sample](https://github.com/ruffle-rs/ruffle/files/7684405/methods.zip)

```
String.prototype.toLowerCase = function()
{
   return "A1";
};
var s = "HeLLo";
trace(s.toLowerCase());
```

Flash Player traces "A1" but Ruffle traces "hello".

The same issue happens to `LoadVars.prototype.getBytesLoaded` and `Date.prototype.getFullYear` (see the attached sample) and likely for all of the methods of in-built objects.

**Comment by adrian17** [comment]
Yeah, looks like we have a `READ_ONLY` flag set on some prototypes where it shouldn't be. I'll try to fix it, at least partially.

**Comment by adrian17** [comment]
These cases (and a couple more) should be now working.
That said, this doesn't cover all the builtins and I'm 100% we still have some attribute mismatches with the official FP, so I'm gonna rename this issue and keep it open.
